src/main/java/org/folio/rmapi/RMAPIService.java

```diff
@@ -0,0 +1,113 @@
+package org.folio.rmapi;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.rmapi.exception.RMAPIResourceNotFoundException;
+import org.folio.rmapi.exception.RMAPIResultsProcessingException;
+import org.folio.rmapi.exception.RMAPIServiceException;
+import org.folio.rmapi.exception.RMAPIUnAuthorizedException;
+
+import java.util.concurrent.CompletableFuture;
+
+public class RMAPIService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RMAPIService.class);
+  private static final String HTTP_HEADER_CONTENT_TYPE = "Content-type";
+  private static final String APPLICATION_JSON = "application/json";
+  private static final String HTTP_HEADER_ACCEPT = "Accept";
+  private static final String RMAPI_API_KEY = "X-Api-Key";
+
+  private static final String JSON_RESPONSE_ERROR = "Error processing RMAPI Response";
+  private static final String INVALID_RMAPI_RESPONSE = "Invalid RMAPI response";
+
+  private String customerId;
+  private String apiKey;
+  private String baseURI;
+
+  private Vertx vertx;
+
+  public RMAPIService(String customerId, String apiKey, String baseURI, Vertx vertx) {
+    this.customerId = customerId;
+    this.apiKey = apiKey;
+    this.baseURI = baseURI;
+    this.vertx = vertx;
+  }
+
+  private <T> CompletableFuture<T> getRequest(String query, Class<T> clazz) {
+
+    CompletableFuture<T> future = new CompletableFuture<>();
+
+    HttpClient httpClient = vertx.createHttpClient();
+
+    final HttpClientRequest request = httpClient.getAbs(query);
+
+    request.headers().add(HTTP_HEADER_ACCEPT, APPLICATION_JSON);
+    request.headers().add(HTTP_HEADER_CONTENT_TYPE, APPLICATION_JSON);
+    request.headers().add(RMAPI_API_KEY, apiKey);
+
+    LOG.info("RMAPI Service absolute URL is" + request.absoluteURI());
+
+    request.handler(response -> response.bodyHandler(body -> {
+      if (response.statusCode() == 200) {
```

Comment:
I think you can close httpClient in the first statement of handler and don't do this for 2 different if branches
see how similar things done with connection (2nd code block):
https://vertx.io/docs/guide-for-java-devs/#_wiki_verticle_initialization_phases